src/Features/LanguageServer/Protocol/Handler/Definitions/AbstractGoToDefinitionHandler.cs

```diff
@@ -62,7 +62,7 @@ public AbstractGoToDefinitionHandler(IMetadataAsSourceFileService metadataAsSour
                 {
                     if (!typeOnly || symbol is ITypeSymbol)
                     {
-                        var declarationFile = await _metadataAsSourceFileService.GetGeneratedFileAsync(document.Project, symbol, signaturesOnly: true, allowDecompilation: false, cancellationToken).ConfigureAwait(false);
+                        var declarationFile = await _metadataAsSourceFileService.GetGeneratedFileAsync(document.Project, symbol, signaturesOnly: false, allowDecompilation: true, cancellationToken).ConfigureAwait(false);
```

Comment:
Just remembered - should we instead pass the 'navigate to decompiled source option' here?
![image](https://user-images.githubusercontent.com/5749229/152232056-17b2123a-5e08-4934-ba8e-6ff74327d6de.png)

Reply:
The decompilation service checks that internally, and ignores the `allowDecompilation` parameter if the option is off.

Reply:
ah got it, lgtm then!